### ISSUE TITLE
Filtering results based on other fields with readonly fields

### ISSUE THREAD

**Issue by robvdl** [yourlabs/django-autocomplete-light#1217]
I have an admin class that once created a foreign key becomes readonly. I use this pattern quite a lot:

```python
    def get_readonly_fields(self, request, obj=None):
        return ["farm"] if obj else []
```

The "farm" foreign key becomes readonly when editing existing records only.

Now when using that in combination with forward=["farm"] on the form field, off course that only works if the form field isn't readonly. Otherwise farm isn't forwarded.

It would be nice that forward= would also work on readonly fields.

I'll see if I can code around it for now somehow, but it would definitely be nice if forward=["farm"] would also work for readonly fields on the admin class.

**Comment by jpic** [comment]
I think the easiest is to add an hidden input with the continent, does that work for you ?

**Comment by robvdl** [comment]
Yeah I was trying something along those lines earlier, I realised that readonly fields probably aren't form fields, so I must add a hidden field.

I'll have another go at it, it gets tricky because I also was using this model as a tabular inline and for inlines you have another problem that the inline doesn't know it's parent yet until it is saved once and has an id.

Happy to close this issue as I realise it's probably not really in scope, but an interesting thing to run into.
